Fix occasional sync failure

```
org.gradle.api.internal.tasks.DefaultTaskContainer$TaskCreationException: Could not create task ':modules:budget:encryption:calculateProjectTree'.
	at org.gradle.api.internal.tasks.DefaultTaskContainer.taskCreationException(DefaultTaskContainer.java:729)
	at org.gradle.api.internal.tasks.DefaultTaskContainer.access$600(DefaultTaskContainer.java:78)
	at org.gradle.api.internal.tasks.DefaultTaskContainer$TaskCreatingProvider.domainObjectCreationException(DefaultTaskContainer.java:721)
	at org.gradle.api.internal.DefaultNamedDomainObjectCollection$AbstractDomainObjectCreatingProvider.tryCreate(DefaultNamedDomainObjectCollection.java:1014)
	at org.gradle.api.internal.tasks.DefaultTaskContainer$TaskCreatingProvider.access$1401(DefaultTaskContainer.java:668)
	at org.gradle.api.internal.tasks.DefaultTaskContainer$TaskCreatingProvider$1.run(DefaultTaskContainer.java:694)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:30)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:27)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:67)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:60)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:167)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:60)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:48)
	at org.gradle.api.internal.tasks.DefaultTaskContainer$TaskCreatingProvider.tryCreate(DefaultTaskContainer.java:690)
	at org.gradle.api.internal.DefaultNamedDomainObjectCollection$AbstractDomainObjectCreatingProvider.calculateOwnValue(DefaultNamedDomainObjectCollection.java:988)
	at org.gradle.api.internal.provider.AbstractMinimalProvider.calculateValue(AbstractMinimalProvider.java:117)
	at org.gradle.api.internal.provider.Collectors$ElementFromProvider.collectEntries(Collectors.java:105)
	at org.gradle.api.internal.provider.Collectors$TypedCollector.collectEntries(Collectors.java:371)
	at org.gradle.api.internal.provider.Collectors$TypedCollector.collectInto(Collectors.java:366)
	at org.gradle.api.internal.collections.SortedSetElementSource.realize(SortedSetElementSource.java:144)
	at org.gradle.api.internal.collections.SortedSetElementSource.realizePending(SortedSetElementSource.java:122)
	at org.gradle.api.internal.collections.SortedSetElementSource.iterator(SortedSetElementSource.java:81)
	at org.gradle.api.internal.DefaultDomainObjectCollection.iterator(DefaultDomainObjectCollection.java:152)
	at org.gradle.api.internal.tasks.DefaultTaskContainer.iterator(DefaultTaskContainer.java:634)
	at java.base/java.lang.Iterable.forEach(Iterable.java:74)
	at com.intellij.gradle.toolingExtension.impl.model.warmUp.GradleTaskWarmUpService.lambda$buildAll$2(GradleTaskWarmUpService.java:46)
	at com.intellij.gradle.toolingExtension.impl.util.GradleResultUtil.lambda$runOrRetryOnce$0(GradleResultUtil.java:34)
	at com.intellij.gradle.toolingExtension.impl.util.GradleResultUtil.runOrRetryOnce(GradleResultUtil.java:16)
	at com.intellij.gradle.toolingExtension.impl.util.GradleResultUtil.runOrRetryOnce(GradleResultUtil.java:33)
	at com.intellij.gradle.toolingExtension.impl.model.warmUp.GradleTaskWarmUpService.buildAll(GradleTaskWarmUpService.java:45)
	at com.intellij.gradle.toolingExtension.impl.modelBuilder.ExtraModelBuilder.buildServiceModel(ExtraModelBuilder.java:87)
	at com.intellij.gradle.toolingExtension.impl.modelBuilder.ExtraModelBuilder.buildAll(ExtraModelBuilder.java:67)
	at com.intellij.gradle.toolingExtension.impl.modelBuilder.ExtraModelBuilder.buildAll(ExtraModelBuilder.java:47)
	at org.gradle.tooling.provider.model.internal.DefaultToolingModelBuilderRegistry$BuilderWithParameter.build(DefaultToolingModelBuilderRegistry.java:291)
	at org.gradle.tooling.provider.model.internal.DefaultToolingModelBuilderRegistry$UserCodeAssigningBuilder.lambda$build$0(DefaultToolingModelBuilderRegistry.java:378)
	at org.gradle.internal.code.DefaultUserCodeApplicationContext$CurrentApplication.reapply(DefaultUserCodeApplicationContext.java:110)
	at org.gradle.tooling.provider.model.internal.DefaultToolingModelBuilderRegistry$UserCodeAssigningBuilder.build(DefaultToolingModelBuilderRegistry.java:378)
	at org.gradle.tooling.provider.model.internal.DefaultToolingModelBuilderRegistry$LockSingleProjectBuilder.lambda$build$0(DefaultToolingModelBuilderRegistry.java:308)
	at org.gradle.api.internal.project.DefaultProjectStateRegistry$ProjectStateImpl.lambda$fromMutableState$2(DefaultProjectStateRegistry.java:458)
	at org.gradle.internal.work.DefaultWorkerLeaseService.withReplacedLocks(DefaultWorkerLeaseService.java:359)
	at org.gradle.api.internal.project.DefaultProjectStateRegistry$ProjectStateImpl.fromMutableState(DefaultProjectStateRegistry.java:458)
	at org.gradle.tooling.provider.model.internal.DefaultToolingModelBuilderRegistry$LockSingleProjectBuilder.build(DefaultToolingModelBuilderRegistry.java:308)
	at org.gradle.tooling.provider.model.internal.DefaultToolingModelBuilderRegistry$BuildOperationWrappingBuilder$1.call(DefaultToolingModelBuilderRegistry.java:341)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:210)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:205)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:67)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:60)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:167)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:60)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:54)
	at org.gradle.tooling.provider.model.internal.DefaultToolingModelBuilderRegistry$BuildOperationWrappingBuilder.build(DefaultToolingModelBuilderRegistry.java:338)
	at org.gradle.internal.build.DefaultBuildToolingModelController$AbstractToolingScope.getModel(DefaultBuildToolingModelController.java:83)
	at org.gradle.internal.buildtree.DefaultBuildTreeModelCreator$DefaultBuildTreeModelController.getModelForScope(DefaultBuildTreeModelCreator.java:146)
	at org.gradle.internal.buildtree.DefaultBuildTreeModelCreator$DefaultBuildTreeModelController.access$300(DefaultBuildTreeModelCreator.java:70)
	at org.gradle.internal.buildtree.DefaultBuildTreeModelCreator$DefaultBuildTreeModelController$1.call(DefaultBuildTreeModelCreator.java:86)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:210)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:205)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:67)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:60)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:167)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:60)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:54)
	at org.gradle.internal.buildtree.DefaultBuildTreeModelCreator$DefaultBuildTreeModelController.getModel(DefaultBuildTreeModelCreator.java:81)
	at org.gradle.tooling.internal.provider.runner.DefaultBuildController.getModel(DefaultBuildController.java:104)
	at org.gradle.tooling.internal.consumer.connection.ParameterAwareBuildControllerAdapter.getModel(ParameterAwareBuildControllerAdapter.java:40)
	at org.gradle.tooling.internal.consumer.connection.UnparameterizedBuildController.getModel(UnparameterizedBuildController.java:116)
	at org.gradle.tooling.internal.consumer.connection.NestedActionAwareBuildControllerAdapter.getModel(NestedActionAwareBuildControllerAdapter.java:32)
	at org.gradle.tooling.internal.consumer.connection.UnparameterizedBuildController.findModel(UnparameterizedBuildController.java:100)
	at org.gradle.tooling.internal.consumer.connection.NestedActionAwareBuildControllerAdapter.findModel(NestedActionAwareBuildControllerAdapter.java:32)
	at org.gradle.tooling.internal.consumer.connection.UnparameterizedBuildController.findModel(UnparameterizedBuildController.java:84)
	at org.gradle.tooling.internal.consumer.connection.NestedActionAwareBuildControllerAdapter.findModel(NestedActionAwareBuildControllerAdapter.java:32)
	at com.intellij.gradle.toolingExtension.impl.util.GradleModelProviderUtil.lambda$buildModelsInParallel$e1e45430$1(GradleModelProviderUtil.java:50)
	at org.gradle.tooling.internal.consumer.connection.NestedActionAwareBuildControllerAdapter$1.get(NestedActionAwareBuildControllerAdapter.java:52)
	at org.gradle.internal.buildtree.IntermediateBuildActionRunner$NestedAction.run(IntermediateBuildActionRunner.java:118)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:67)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:60)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:167)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:60)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$QueueWorker.execute(DefaultBuildOperationExecutor.java:161)
	at org.gradle.internal.operations.DefaultBuildOperationQueue$WorkerRunnable.runOperation(DefaultBuildOperationQueue.java:272)
	at org.gradle.internal.operations.DefaultBuildOperationQueue$WorkerRunnable.doRunBatch(DefaultBuildOperationQueue.java:253)
	at org.gradle.internal.operations.DefaultBuildOperationQueue$WorkerRunnable.lambda$runBatch$1(DefaultBuildOperationQueue.java:226)
	at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:263)
	at org.gradle.internal.work.DefaultWorkerLeaseService.runAsWorkerThread(DefaultWorkerLeaseService.java:127)
	at org.gradle.internal.operations.DefaultBuildOperationQueue$WorkerRunnable.runBatch(DefaultBuildOperationQueue.java:224)
	at org.gradle.internal.operations.DefaultBuildOperationQueue$WorkerRunnable.run(DefaultBuildOperationQueue.java:192)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:48)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: org.gradle.api.internal.tasks.DefaultTaskContainer$TaskCreationException: Could not create task ':collateProjectLinks'.
	at org.gradle.api.internal.tasks.DefaultTaskContainer.taskCreationException(DefaultTaskContainer.java:729)
	at org.gradle.api.internal.tasks.DefaultTaskContainer.access$600(DefaultTaskContainer.java:78)
	at org.gradle.api.internal.tasks.DefaultTaskContainer$TaskCreatingProvider.domainObjectCreationException(DefaultTaskContainer.java:721)
	at org.gradle.api.internal.DefaultNamedDomainObjectCollection$AbstractDomainObjectCreatingProvider.tryCreate(DefaultNamedDomainObjectCollection.java:1014)
	at org.gradle.api.internal.tasks.DefaultTaskContainer$TaskCreatingProvider.access$1401(DefaultTaskContainer.java:668)
	at org.gradle.api.internal.tasks.DefaultTaskContainer$TaskCreatingProvider$1.run(DefaultTaskContainer.java:694)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:30)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:27)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:67)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:60)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:167)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:60)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:48)
	at org.gradle.api.internal.tasks.DefaultTaskContainer$TaskCreatingProvider.tryCreate(DefaultTaskContainer.java:690)
	at org.gradle.api.internal.DefaultNamedDomainObjectCollection$AbstractDomainObjectCreatingProvider.calculateOwnValue(DefaultNamedDomainObjectCollection.java:988)
	at org.gradle.api.internal.provider.AbstractMinimalProvider.calculateOwnPresentValue(AbstractMinimalProvider.java:82)
	at org.gradle.api.internal.provider.AbstractMinimalProvider.get(AbstractMinimalProvider.java:102)
	at org.gradle.api.internal.DefaultNamedDomainObjectCollection$AbstractDomainObjectCreatingProvider.get(DefaultNamedDomainObjectCollection.java:974)
	at actual.diagrams.tasks.CalculateProjectTreeTask$Companion$register$1$1$1.execute(CalculateProjectTreeTask.kt:85)
	at actual.diagrams.tasks.CalculateProjectTreeTask$Companion$register$1$1$1.execute(CalculateProjectTreeTask.kt:84)
	at org.gradle.api.internal.DefaultMutationGuard$1.execute(DefaultMutationGuard.java:66)
	at org.gradle.api.internal.DefaultMutationGuard$1.execute(DefaultMutationGuard.java:66)
	at org.gradle.internal.code.DefaultUserCodeApplicationContext$CurrentApplication$1.execute(DefaultUserCodeApplicationContext.java:124)
	at org.gradle.api.internal.DefaultCollectionCallbackActionDecorator$BuildOperationEmittingAction$1.run(DefaultCollectionCallbackActionDecorator.java:110)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:30)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:27)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:67)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:60)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:167)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:60)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:48)
	at org.gradle.api.internal.DefaultCollectionCallbackActionDecorator$BuildOperationEmittingAction.execute(DefaultCollectionCallbackActionDecorator.java:107)
	at org.gradle.internal.ImmutableActionSet$SetWithManyActions.execute(ImmutableActionSet.java:329)
	at org.gradle.api.internal.DefaultDomainObjectCollection.doAdd(DefaultDomainObjectCollection.java:280)
	at org.gradle.api.internal.DefaultNamedDomainObjectCollection.doAdd(DefaultNamedDomainObjectCollection.java:125)
	at org.gradle.api.internal.DefaultNamedDomainObjectCollection$AbstractDomainObjectCreatingProvider.tryCreate(DefaultNamedDomainObjectCollection.java:1008)
	... 87 more
Caused by: java.lang.NullPointerException: Cannot invoke "org.gradle.api.Named.getName()" because "object" is null
	at org.gradle.api.Named$Namer.determineName(Named.java:46)
	at org.gradle.api.Named$Namer.determineName(Named.java:40)
	at org.gradle.api.internal.DefaultNamedDomainObjectCollection.doAdd(DefaultNamedDomainObjectCollection.java:123)
	at org.gradle.api.internal.DefaultNamedDomainObjectCollection$AbstractDomainObjectCreatingProvider.tryCreate(DefaultNamedDomainObjectCollection.java:1008)
	... 119 more
```

--- a/build-logic/src/main/kotlin/actual/diagrams/ModuleType.kt
+++ b/build-logic/src/main/kotlin/actual/diagrams/ModuleType.kt
@@ -1,35 +1,65 @@
 package actual.diagrams
 
+import actual.diagrams.ModuleType.AndroidApp
+import actual.diagrams.ModuleType.AndroidCompose
+import actual.diagrams.ModuleType.AndroidHilt
+import actual.diagrams.ModuleType.AndroidLibrary
+import actual.diagrams.ModuleType.AndroidResources
+import actual.diagrams.ModuleType.AndroidViewModel
+import actual.diagrams.ModuleType.Jvm
+import actual.diagrams.ModuleType.Multiplatform
+import actual.diagrams.ModuleType.Other
 import guru.nidi.graphviz.attribute.Color
 import org.gradle.api.Project
 
-enum class ModuleType(val label: String, val color: Color) {
-  AndroidApp(string = "App", fill = "#FF5555"), // red
-  AndroidViewModel(string = "ViewModel", fill = "#F5A6A6"), // pink
-  AndroidHilt(string = "Hilt", fill = "#FCB103"), // orange
-  AndroidCompose(string = "Compose", fill = "#FFFF55"), // yellow
-  AndroidLibrary(string = "Android", fill = "#55FF55"), // green
-  AndroidResources(string = "Resources", fill = "#00FFFF"), // cyan
-  Multiplatform(string = "Multiplatform", fill = "#9D8DF1"), // indigo
-  Jvm(string = "JVM", fill = "#8000FF"), // violet
-  Other(string = "Other", fill = "#808080"), // grey
-  ;
+enum class ModuleType {
+  AndroidApp,
+  AndroidViewModel,
+  AndroidHilt,
+  AndroidCompose,
+  AndroidLibrary,
+  AndroidResources,
+  Multiplatform,
+  Jvm,
+  Other,
+}
 
-  constructor(string: String, fill: String) : this(string, Color.rgb(fill).fill())
-
-  companion object {
-    fun find(project: Project): ModuleType = with(project.plugins) {
-      when {
-        hasPlugin("com.android.application") -> AndroidApp
-        hasPlugin("actual.module.viewmodel") -> AndroidViewModel
-        hasPlugin("actual.module.hilt") -> AndroidHilt
-        hasPlugin("actual.module.compose") -> AndroidCompose
-        hasPlugin("actual.module.android") -> AndroidLibrary
-        hasPlugin("actual.module.resources") -> AndroidResources
-        hasPlugin("actual.module.multiplatform") -> Multiplatform
-        hasPlugin("actual.module.jvm") -> Jvm
-        else -> Other
-      }
-    }
+internal fun ModuleType(project: Project) = with(project.pluginManager) {
+  when {
+    hasPlugin("com.android.application") -> AndroidApp
+    hasPlugin("actual.module.viewmodel") -> AndroidViewModel
+    hasPlugin("actual.module.hilt") -> AndroidHilt
+    hasPlugin("actual.module.compose") -> AndroidCompose
+    hasPlugin("actual.module.android") -> AndroidLibrary
+    hasPlugin("actual.module.resources") -> AndroidResources
+    hasPlugin("actual.module.multiplatform") -> Multiplatform
+    hasPlugin("actual.module.jvm") -> Jvm
+    else -> Other
   }
 }
+
+internal val ModuleType.label: String
+  get() = when (this) {
+    AndroidApp -> "App"
+    AndroidViewModel -> "ViewModel"
+    AndroidHilt -> "Hilt"
+    AndroidCompose -> "Compose"
+    AndroidLibrary -> "Android"
+    AndroidResources -> "Resources"
+    Multiplatform -> "Multiplatform"
+    Jvm -> "JVM"
+    Other -> "Other"
+  }
+
+internal val ModuleType.color: Color
+  get() = when (this) {
+    AndroidApp -> "#FF5555" // red
+    AndroidViewModel -> "#F5A6A6" // pink
+    AndroidHilt -> "#FCB103" // orange
+    AndroidCompose -> "#FFFF55" // yellow
+    AndroidLibrary -> "#55FF55" // green
+    AndroidResources -> "#00FFFF" // cyan
+    Multiplatform -> "#9D8DF1" // indigo
+    Jvm -> "#8000FF" // violet
+    Other -> "#808080" // grey
+  }.let(Color::rgb).fill()

--- a/build-logic/src/main/kotlin/actual/diagrams/ModuleType.kt
+++ b/build-logic/src/main/kotlin/actual/diagrams/ModuleType.kt
@@ -24,7 +24,7 @@ enum class ModuleType {
   Other,
 }
 
-internal fun ModuleType(project: Project) = with(project.pluginManager) {
+internal fun moduleType(project: Project): ModuleType = with(project.pluginManager) {
   when {
     hasPlugin("com.android.application") -> AndroidApp
     hasPlugin("actual.module.viewmodel") -> AndroidViewModel

--- a/build-logic/src/main/kotlin/actual/diagrams/tasks/CollateModuleTypesTask.kt
+++ b/build-logic/src/main/kotlin/actual/diagrams/tasks/CollateModuleTypesTask.kt
@@ -39,7 +39,10 @@ abstract class CollateModuleTypesTask : DefaultTask() {
       }
 
       gradle.projectsEvaluated {
-        val dumpTasks = rootProject.subprojects.mapNotNull(DumpModuleTypeTask::get)
+        val dumpTasks = rootProject
+          .subprojects
+          .toList()
+          .mapNotNull(DumpModuleTypeTask::get)
         task.configure {
           dependsOn(dumpTasks)
           projectTypeFiles.from(dumpTasks.map { it.get().outputFile })

--- a/build-logic/src/main/kotlin/actual/diagrams/tasks/CollateProjectLinksTask.kt
+++ b/build-logic/src/main/kotlin/actual/diagrams/tasks/CollateProjectLinksTask.kt
@@ -65,7 +65,10 @@ abstract class CollateProjectLinksTask : DefaultTask() {
       }
 
       gradle.projectsEvaluated {
-        val dumpTasks = rootProject.subprojects.mapNotNull(DumpProjectLinksTask::get)
+        val dumpTasks = rootProject
+          .subprojects
+          .toList()
+          .mapNotNull(DumpProjectLinksTask::get)
         task.configure {
           dependsOn(dumpTasks)
           projectLinkFiles.from(dumpTasks.map { it.get().outputFile })

--- a/build-logic/src/main/kotlin/actual/diagrams/tasks/DumpModuleTypeTask.kt
+++ b/build-logic/src/main/kotlin/actual/diagrams/tasks/DumpModuleTypeTask.kt
@@ -47,7 +47,7 @@ abstract class DumpModuleTypeTask : DefaultTask() {
 
       target.afterEvaluate {
         task.configure {
-          type.set(ModuleType.find(target))
+          type.set(ModuleType(target))
         }
       }
 

--- a/build-logic/src/main/kotlin/actual/diagrams/tasks/DumpModuleTypeTask.kt
+++ b/build-logic/src/main/kotlin/actual/diagrams/tasks/DumpModuleTypeTask.kt
@@ -1,6 +1,7 @@
 package actual.diagrams.tasks
 
 import actual.diagrams.ModuleType
+import actual.diagrams.moduleType
 import org.gradle.api.DefaultTask
 import org.gradle.api.Project
 import org.gradle.api.UnknownTaskException
@@ -47,7 +48,7 @@ abstract class DumpModuleTypeTask : DefaultTask() {
 
       target.afterEvaluate {
         task.configure {
-          type.set(ModuleType(target))
+          type.set(moduleType(target))
         }
       }
 

--- a/build-logic/src/main/kotlin/actual/diagrams/tasks/GenerateDotFileTask.kt
+++ b/build-logic/src/main/kotlin/actual/diagrams/tasks/GenerateDotFileTask.kt
@@ -1,6 +1,8 @@
 package actual.diagrams.tasks
 
 import actual.diagrams.ModuleType
+import actual.diagrams.color
+import actual.diagrams.label
 import guru.nidi.graphviz.attribute.Attributes
 import guru.nidi.graphviz.attribute.Color
 import guru.nidi.graphviz.attribute.Font

